### DISCUSSION TITLE
QTable filtering when cell value is custom

### DISCUSSION THREAD

**Discussion by psuplat** [quasarframework/quasar#14699] [General - Components / Directives / etc] [answered]
Hey, I have a QTable with a customised cell content using slot. That cell displays combined data from 3 fields. The problem I have is that the values in that field does not seem to be taken into account when filtering is done. Am I missing something here?

This is the definition of that particular header:
`{ name: 'reported_by', label: 'Name/Department', field: '', align: 'left', sortable: true }`

How the sample data looks like:
```
{
    "id": "1123",
    "department_id": "1",
    "location_id": "1",
    "type_id": "4",
    "incident_date": "2022-10-15",
    "incident_time": "15:20:00",
    "first_name": "John",
    "last_name": "Doe",
    "department_name": "IT",
    "brief_description": "zzzzz",
    "immediate_actions": "none taken",
    "status": "OPEN",
}
```

and this is how the slot is configured:
```
<template v-slot:body-cell-reported_by="props">
    <q-td :props="props">
        {{props.row.first_name}} {{props.row.last_name}}<br /><span class="text-grey-7">{{props.row.department_name}}</span>
    </q-td>
</template>
```

if I search for _zzz,_ which is part of description column, the filter picks it up:
![obraz](https://user-images.githubusercontent.com/14207860/197502257-cff3dd58-6ab5-46e1-98ea-795857fde59f.png)

but if I search for name or department I get nothing:
![obraz](https://user-images.githubusercontent.com/14207860/197502181-328418c0-1fec-43d3-abf0-475cfbc3c877.png)

How can I make the filter pick up the value from custom cell slot?

**Comment by MilosPaunovic** [comment, marked as answer]
You need to implement custom `filter-method` as well if you want to filter on that column.
